### PR TITLE
[indexer] Upgrade sf-stream proto generation

### PR DESCRIPTION
**Pull request by bowenyang007**
### Description
* Minor typo fixes: 
  * ed255198 -> ed25519
  * DELETE_MODULE -> WRITE_MODULE
* Switch from protobuf to prost- a required trait for SF
 * enable serialize support for prost-types by leveraging buf build pipeline
* Move proto file to a common crate in crates/aptos-protos

### Test Plan
cargo test
firehose still works

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2728)

### CODE REVIEW

File: crates/aptos-protos/proto/timestamp.proto [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package aptos.util.timestamp;
+
+message Timestamp {
+  // Represents seconds of UTC time since Unix epoch
+  // 1970-01-01T00:00:00Z. Must be from 0001-01-01T00:00:00Z to
+  // 9999-12-31T23:59:59Z inclusive.
```
Comment (jjleng) [inline crates/aptos-protos/proto/timestamp.proto:11]: What does this mean
`Must be from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59Z inclusive.`

File: crates/aptos-protos/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Aptos Protobufs
+
+This is a simple crate for working with the [Aptos](https://aptos.org) protobufs
+
+
+## Generating protos
+We use [buf](https://docs.buf.build/introduction) to generate protos.
+
+```bash
+brew install buf
```
[the remark this answers is not in the document]
Comment (jjleng) [reply crates/aptos-protos/README.md:10]: Can we put this in dev_setup.sh?

**Review by jjleng — APPROVED** [review]

File: crates/aptos-protos/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+homepage = "https://aptoslabs.com"
+license = "Apache-2.0"
+publish = false
+edition = "2018"
+
+[dependencies]
+bytes = "1.1.0"
+chrono = { version = "0.4.19", default-features = false, features = ["serde"] }
+prost = "0.10.4"
+pbjson = "0.4.0"
```
[the remark this answers is not in the document]
Comment (CapCap) [reply crates/aptos-protos/Cargo.toml]: we do, it did lol (eventually). The git hooks we have are a constant source of pain for me lol

File: crates/aptos-protos/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (CapCap) [reply crates/aptos-protos/README.md:10]: :'(

File: crates/aptos-protos/proto/timestamp.proto (hunk 1 above)
Comment (CapCap) [reply crates/aptos-protos/proto/timestamp.proto:11]: this is the standard of values that are allowed (this is stolen from original google definition); in a full u64, you can express post-heat-death-of-universe times, which break a LOT of downstream consumers which don't; year 9999 is a great end cap on that limit

File: crates/aptos-protos/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (CapCap) [reply crates/aptos-protos/README.md]: I could @jjleng added installation instructions for mac + linux; can I do the dev_stup for this in a follow up PR?

**Pull request merged**